MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -86,8 +85,18 @@ - (BOOL)loadLogsWithGroupId:(NSString *)groupId
   logsAvailable = logs.count > 0;
   if (logsAvailable) {
     batchId = MS_UUID_STRING;
-    [self.batches setObject:(NSArray<NSString *> * _Nonnull)[logs allKeys]
-                     forKey:[groupId stringByAppendingString:batchId]];
+    
+    // Update the logs in the DB with the batchId
+    NSString *logIdsForBatch = [[logs allKeys] componentsJoinedByString:@"','"];
+    
+    NSString *updateLogsQuery = [NSString stringWithFormat:@"UPDATE %@ SET %@ = '%@' WHERE %@ IN ('%@')", kMSLogTableName, kMSBatchIdColumnName, batchId, kMSIdColumnName, logIdsForBatch];
```

Comment:
Don't we need retries here?
What will happen if update fails? Current implementation will resend logs.